Lib/zipfile/__init__.py

```diff
@@ -508,6 +518,24 @@ def _decodeExtra(self):
                 except struct.error:
                     raise BadZipFile(f"Corrupt zip64 extra field. "
                                      f"{field} not found.") from None
+            elif tp == 0x7075:
+                data = extra[4:ln+4]
+                # Unicode Path Extra Field
+                try:
+                    up_version, up_name_crc = unpack('<BL', data[:5])
+                    if up_version == 1 and up_name_crc == self.orig_filename_crc:
```

Comment:
Please check that it will be enough to calculate crc32 on the fly.

```suggestion
                    if up_version == 1 and up_name_crc == crc32(self.filename):
```

Reply:
In my opinion, `orig_filename_crc` is necessary. `filename` in ZipInfo is the decoded value from ZipFile. But `orig_filename_crc` is the result of the CRC on `filename` before decoding. So it should be stored separately.

Reply:
@yeojin-dev 

Okay, how about passing the orig_filename_crc to _decodeExtra?

https://github.com/python/cpython/blob/c75e016665de2cdd8705b32687fa8937a1c27796/Lib/zipfile/__init__.py#L1436-L1463

Looks like `_decodeExtra` is only used at this moment. WDYT?

Reply:
That's a good idea! I will update. :)